src/api/controller/api/publishedDataset.js

```diff
@@ -10,6 +15,26 @@ export default async (ctx) => {
 
     ctx.body = {
         total,
-        data,
+        data: data.map(doc => ({
+            ...doc.versions.slice(-1)[0],
+            uri: doc.uri,
+        })),
     };
 };
+
+export const editResource = async (ctx) => {
+    const newVersion = ctx.request.body;
+    const resource = await ctx.publishedDataset.findByUri(newVersion.uri);
+    if (!resource) {
+        ctx.status = 403;
+        ctx.body = `Forbidden: No older version has been found for document ${newVersion.uri}`;
```

Comment:
`Document not found` would be better imo. This message suggests the document might exist